app/src/test/java/com/github/se/signify/model/stats/StatsViewModelTest.kt

```diff
@@ -2,136 +2,221 @@ package com.github.se.signify.model.stats
-    statsViewModel.getCompletedChallengeStats()
-    verify(statsRepositoryFirestore).getCompletedChallengeStats(eq(userId), any(), any())
+  fun `updateIntFunctions should handle failure correctly`() {
+    mockStatsRepository.shouldSucceed = false
+
+    updateAction()
+
+    assertTrue(mockStatsRepository.wasMethodCalled(updateStatName))
+
+    mockStatsRepository.shouldSucceed = true
+    getAction()
+
+    assertEquals(expectedValueGet, stateFlowValue())
   }
 
   @Test
-  fun getCreatedChallengeStatsShouldCallRepositoryGetCreatedChallengeStats() {
-    statsViewModel.getCreatedChallengeStats()
-    verify(statsRepositoryFirestore).getCreatedChallengeStats(eq(userId), any(), any())
+  fun getLettersLearnedShouldHandleSuccessCorrectly() {
+    statsViewModel.getLettersLearned()
+
+    assertTrue(mockStatsRepository.wasMethodCalled("getLettersLearned"))
+
+    assertEquals(initialStats.lettersLearned, statsViewModel.lettersLearned.value)
   }
 
   @Test
-  fun getWonChallengeStatsShouldCallRepositoryGetWonChallengeStats() {
-    statsViewModel.getWonChallengeStats()
-    verify(statsRepositoryFirestore).getWonChallengeStats(eq(userId), any(), any())
+  fun getLettersLearnedShouldHandleFailureCorrectly() {
+    mockStatsRepository.shouldSucceed = false
+
+    statsViewModel.getLettersLearned()
+
+    assertTrue(mockStatsRepository.wasMethodCalled("getLettersLearned"))
+
+    assertEquals(emptyList<Char>(), statsViewModel.lettersLearned.value)
   }
 
   @Test
-  fun updateLettersLearnedShouldCallRepositoryUpdateLettersLearned() {
+  fun updateLettersLearnedShouldHandleSuccessCorrectly() {
     val newLetter = 'D'
+
     statsViewModel.updateLettersLearned(newLetter)
-    verify(statsRepositoryFirestore).updateLettersLearned(eq(userId), eq(newLetter), any(), any())
-  }
 
-  @Test
-  fun updateEasyExerciseStatsShouldCallRepositoryUpdateEasyExerciseStats() {
-    statsViewModel.updateEasyExerciseStats()
-    verify(statsRepositoryFirestore).updateEasyExerciseStats(eq(userId), any(), any())
-  }
+    assertTrue(mockStatsRepository.wasMethodCalled("updateLettersLearned"))
 
-  @Test
-  fun updateMediumExerciseStatsShouldCallRepositoryUpdateMediumExerciseStats() {
-    statsViewModel.updateMediumExerciseStats()
-    verify(statsRepositoryFirestore).updateMediumExerciseStats(eq(userId), any(), any())
-  }
+    statsViewModel.getLettersLearned()
 
-  @Test
-  fun updateHardExerciseStatsShouldCallRepositoryUpdateHardExerciseStats() {
-    statsViewModel.updateHardExerciseStats()
-    verify(statsRepositoryFirestore).updateHardExerciseStats(eq(userId), any(), any())
+    assertEquals(initialStats.lettersLearned + newLetter, statsViewModel.lettersLearned.value)
   }
 
   @Test
-  fun updateDailyQuestStatsShouldCallRepositoryUpdateDailyQuestStats() {
-    statsViewModel.updateDailyQuestStats()
-    verify(statsRepositoryFirestore).updateDailyQuestStats(eq(userId), any(), any())
-  }
+  fun updateLettersLearnedShouldHandleFailureCorrectly() {
```

Comment:
The updateLettersLearned tests do not account for duplicate letters or cases where the new letter is invalid (e.g., a non-alphabetic character). Suggestion: Add tests to handle such cases and verify that the repository either rejects invalid updates or ensures the integrity of the lettersLearned list.